Public/Src/Engine/Processes/AnyBuildShimInfo.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using BuildXL.Utilities;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// Configuration for shim execution in place of child processes executing in a build sandbox.
+    /// </summary>
+    public sealed class AnyBuildShimInfo
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public AnyBuildShimInfo(AbsolutePath substituteProcessExecutionShimPath, bool shimAllProcesses, IReadOnlyCollection<ShimProcessMatch> processMatches)

```

Comment:
Do you allow `processMatches` to be null?

Reply:
No, will add an assert on that

---
In reply to: [282685088](https://github.com/microsoft/BuildXL/pull/279#discussion_r282685088) [](ancestors = 282685088)